weavertest/init.go

```diff
@@ -28,48 +28,57 @@ import (
 	"github.com/ServiceWeaver/weaver/runtime/logging"
 )
 
-// Mode controls where components live and how their methods are invoked.
-type Mode uint8
-
-const (
-	// Local places all components in the same process and uses local procedure calls
-	// for method invocations.
-	// This is the default value.
-	Local Mode = iota
-
-	// RPC places all components in the same process and uses RPCs for method invocations.
-	RPC
-
-	// Multi places all components in different process (unless explicitly colocated)
-	// and uses RPCs for method invocations on remote components and local procedure calls
-	// for method invocations on colocated components.
-	Multi
+// Runner runs user-supplied testing code as a weaver application.
+type Runner struct {
+	multi    bool // Use multiple processes
+	forceRPC bool // Use RPCs even for local calls
+	name     string
+	config   string
+}
+
+var (
+	// Local is a Runner that places all components in the same process
+	// and uses local procedure calls for method invocations.
+	Local = Runner{name: "Local"}
+
+	// RPC is a Runner that places all components in the same process
+	// and uses RPCs for method invocations.
+	RPC = Runner{multi: false, forceRPC: true, name: "RPC"}
+
+	// Multi is a Runner that places all components in different
+	// process (unless explicitly colocated) and uses RPCs for method
+	// invocations on remote components and local procedure calls for
+	// method invocations on colocated components.
+	Multi = Runner{multi: true, name: "Multi"}
 )
 
-// String returns a string representation of a Mode, suitable for using as sub-test names.
-func (m Mode) String() string {
-	switch m {
-	case RPC:
-		return "RPC"
-	case Local:
-		return "Local"
-	case Multi:
-		return "Multi"
-	default:
-		panic(fmt.Sprintf("unknown mode %d", m))
+// AllRunners returns a slice of all builtin weavertest runners.
+func AllRunners() []Runner { return []Runner{Local, RPC, Multi} }
+
+// WithName returns a new Runner with the specified name. It is useful
+// when the runner has been adjusted and is no longer identical to one
+// of the predefined runners.
+func (r Runner) WithName(name string) Runner { r.name = name; return r }
+
+// WithConfig returns a new Runner with the specified Service Weaver
+// configuration. The config value passed here is identical to what
+// might be found in a Service Weaver config file. It can contain
+// application level as well as component level configuration.
+func (r Runner) WithConfig(config string) Runner { r.config = config; return r }
+
+// Name returns the runner name. It is suitable for use as a sub-test or sub-benchmark name.
+func (r Runner) Name() string {
+	if r.name == "" {
+		return "Default"
```

Comment:
nit: would "Undefined", "Unknown"  be a better name? "Default" sounds a little bit misleading (maybe).